pan3d/dataset_builder.py

```diff
@@ -261,77 +281,74 @@ def on_set_array_active(self, array_active, **kwargs):
                     "step": 1,
                 }
             )
-            self.state.expanded_coordinates.append(key),
+            self.state.ui_expanded_coordinates.append(key),
         self.auto_select_coordinates()
-        self.state.dirty("coordinates", "expanded_coordinates")
+        self.state.dirty("da_coordinates", "ui_expanded_coordinates")
+        self.plotter.clear()
+        self.plotter.view_isometric()
 
-    @change("x_array")
-    def on_set_x_array(self, x_array, **kwargs):
-        self.algorithm.x = x_array
+    @change("da_x")
+    def on_set_da_x(self, da_x, **kwargs):
+        self.algorithm.x = da_x
         self.mesh_changed()
```

Comment:
I would group those into a single method.
Something like

```python
@change("da_x", "da_y", "da_z")
def on_coordinate_change(self, da_x, da_y, da_z, **kwargs):
      self.algorithm.x = da_x
      self.algorithm.y = da_y
      self.algorithm.z = da_z
      self.mesh_changed()
```

Reply:
That is much more succinct, thank you. Changed in [41e3c22](https://github.com/Kitware/pan3d/pull/33/commits/41e3c224749c308565d13cd4bc51f0e5e62d7423)